### ISSUE TITLE
Adding Constraint on Alignment Saturation

### ISSUE THREAD

**Issue by fatihkiralioglu** [NVIDIA/tacotron2#357]
Hi,
I have trained a tacotron2 model on a custom dataset and while testing newly generated model, I have come across some input examples where alignment is maximized way before gate_threshold or max decoder steps conditions are satisfied:

> if torch.sigmoid(gate_output.data) > self.gate_threshold:
>                 break
>  elif len(mel_outputs) == self.max_decoder_steps:
>                 print("Warning! Reached max decoder steps")
>                 break

Mel and alignment output plots are like the following for a specific case:

![image](https://user-images.githubusercontent.com/38240476/82890602-1fa27b00-9f55-11ea-970f-a75cd63a81f3.png)

As you can see, the alignment is maximized at about 3000 ms and the synthesis is completed successfully for the input text, however, because gate output score did not reach the threshold, decoder continues and produces redundant synthesis.

What I wonder is whether it is possible to add a new constraint for alignment so that after alignment saturates, inference loop is stopped.

Thanks.

**Comment by CookiePPP** [comment]
@fatihkiralioglu 
https://github.com/CookiePPP/codedump/blob/master/tacotron2-PPP-1.3.0/loss_function.py#L27
https://pytorch.org/docs/stable/nn.html#torch.nn.BCEWithLogitsLoss


You might fix problem by adding
```
        gate_pos_weight = torch.tensor( 10.0 )
        gate_loss = nn.BCEWithLogitsLoss(pos_weight=gate_pos_weight)(gate_out, gate_target)
```
or something similar to [loss function](https://github.com/NVIDIA/tacotron2/blob/master/loss_function.py#L18) during training.

**Comment by rafaelvalle** [comment]
Closing due to inactivity.
